adagrams/game.py

```diff
@@ -1,11 +1,193 @@
+import random
+from collections import Counter
+
+#creat letter pool 
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+
+
 def draw_letters():
-    pass
+    #make a copy of the letter pool
+    user_pool = dict(LETTER_POOL)
+
+    #create a list to sotre user drawn on hand
+    hand = []
+    
+    #use while loop to draw a random letter from letter pool
+    while len(hand) < 10:
+        # get all the letters from letter pool and store in a list
+        letters = list(LETTER_POOL.keys())
+
+        # get a random index from the letters list
+        random_index = random.randint(0, len(letters) -1)
+
+        # get the random letter by using the random index
+        random_letter = letters[random_index]
+        # get the letter counts from the copy of letter pool, everytime draw a letter, the value update -1
+        num = user_pool[random_letter]
+        user_pool[random_letter] = num - 1
+
+        #if no remaining letter in copy of letter pool, skip, and re-draw
+        if user_pool[random_letter] <= 0:
+            continue
+        else:
+            hand.append(random_letter)
+
+    return hand
+    
+'''
+    #use random.choice()
+    # when the user has less 10 letter on hand, 
+    # continue to drawn from the user pool, store in the list
+    # if the remaining letter in user pool is 0,
+    # jump to drawn a new letter.
+    while len(hand) < 10:
+        
+        letter = random.choice(list(user_pool.keys()))
+        
+        num = user_pool[letter]
+
+        user_pool[letter] = num - 1;
+        
+        if user_pool[letter] <= 0:
+            continue
+        else:
+            hand.append(letter)
+'''
+    
+    
+    
 
 def uses_available_letters(word, letter_bank):
-    pass
+    
+    #initialize an empty dictinary to store the frequency of each letter
+    frequency = {}
+
+    #loop through each letter from the letter_bank,
+    #if the letter in frequency{}, value +1
+    #if not, add the letter to frequency{} and value 1
+    for letter in letter_bank:
+        if letter in frequency:
+            frequency[letter] += 1
+        else:
+            frequency[letter] = 1  
+
+    #check if word in letter_bank()
+    #ignore the case sensitive
+    #if the letter in the word,
+    #and the freqency of the letter in word is lees than frequency of letter in letter bank, return True
+    #else return False
+    for letter in word.upper():
+        if letter in frequency.keys():
+            if frequency[letter] > 0:
+                frequency[letter] -= 1
+            else:
+                return False
+        else:
+            return False
```

Comment:
In your for loop, you first check if a letter is in the keys and then you do another check to see if the value is greater than 0. When you do this you nest conditionals which increases the complexity. 

Prefer to simplify the logic by using fewer conditional when possible. How about something like this?

```python
for letter in word.upper():
        if frequency.get(letter, 0) < 1:            
              return False 
        letter_bank_count[char] -= 1

return True
```